lib/common/output.c

```diff
@@ -91,7 +92,38 @@ pcmk__register_format(const char *fmt_name, pcmk__output_factory_t create) {
         formatters = g_hash_table_new_full(crm_str_hash, g_str_equal, NULL, NULL);
     }
 
-    g_hash_table_insert(formatters, strdup(fmt_name), create);
+    if (options != NULL && context != NULL) {
+        char *group_name = crm_strdup_printf("output-%s", name);
+        char *group_desc = crm_strdup_printf("%s output options", name);
+        char *group_help = crm_strdup_printf("Show %s output help", name);
+
+        GOptionGroup *group = g_option_group_new(group_name, group_desc,
+                                                 group_help, NULL, NULL);
+
+        g_option_group_add_entries(group, options);
+        g_option_context_add_group(context, group);
+
+        free(group_name);
+        free(group_desc);
+        free(group_help);
+    }
+
+    g_hash_table_insert(formatters, strdup(name), create);
+    return 0;
+}
+
+int
+pcmk__register_formats(GOptionContext *context, pcmk__supported_format_t *formats) {
+    pcmk__supported_format_t *entry = NULL;
+    int rc = 0;
+
+    for (entry = formats; entry->name != NULL; entry++) {
+        rc = pcmk__register_format(context, entry->name, entry->create, entry->options);
+        if (rc != 0) {
+           return rc;
```

Comment:
I'm not sure what the error handling should be here. Do we really care if a format fails to register (in the sense of should we do anything differently if it does)? I don't think we should bail if a format can't be registered -- the user might not even have selected it.